azurerm/internal/services/postgres/postgresql_flexible_server_database_resource.go

```diff
@@ -0,0 +1,162 @@
+package postgres
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/postgresql/mgmt/2021-06-01/postgresqlflexibleservers"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/postgres/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/postgres/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourcePostgresqlFlexibleServerDatabase() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourcePostgresqlFlexibleServerDatabaseCreate,
+		Read:   resourcePostgresqlFlexibleServerDatabaseRead,
+		Delete: resourcePostgresqlFlexibleServerDatabaseDelete,
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.FlexibleServerDatabaseID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FlexibleServerDatabaseName,
+			},
+
+			"server_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FlexibleServerID,
+			},
+
+			"charset": {
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     validate.DatabaseCharset,
+			},
+
+			"collation": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DatabaseCollation,
+			},
```

Comment:
These two fields are optional. The service team is fixing that they are required currently. This will be release in the next week.

Reply:
Then we should probably wait until next week, there are also some other fixes for end of next week.

Reply:
@yupwei68 Any update on this service change?

Reply:
If this change takes more time on Azure Side, should we make it required for now and change it later to optional?

Reply:
OK